### ISSUE TITLE
Framework/glue code plan

### ISSUE THREAD

**Issue by m-toman** [m-toman/tacorn#13]
I'm currently reworking the general training procedure to more easily enable
- use your own dataset
- pick a pretrained model
- have the whole training inside an external experiment folder
- later on swap out the implementations of Tacotron and the neural vocoder

**Comment by Yeongtae** [comment]
@m-toman Could you tell me the inference time per sec of your wavernn model. In addition, your GPU device.

**Comment by m-toman** [comment]
That is still true, as I haven't really finished any new integration yet.
I'm currently integrating this fork of the fork of the fatchord model: https://github.com/geneing/WaveRNN-Pytorch
which should be a bit faster when using batch synthesis.

Did you have any luck with WaveGlow?

**Comment by Yeongtae** [comment]
@m-toman 
nvidia-tacotron2 and nvidia-waveglow are well optimized.
with a v100 machine can generate 160k samples/sec, 350k samples/sec each.

But waveglow have a problem about reverb.
I'm trying to overcome this problem. 
https://github.com/Yeongtae/tacotron2
https://github.com/Yeongtae/waveglow

**Comment by m-toman** [comment]
Impressive, I also wanted to take a look at their repo but can't jump between them all the time ;).
I've seen in the WaveGlow issues that the training requires lots of memory to achieve a reasonable batch size.

**Comment by Yeongtae** [comment]
So I'm using 8 v100 gpus to train waveglow.

**Comment by Yeongtae** [comment]
@m-toman do you make some results with this repository?
Could you share a sample audio?

**Comment by ZohaibAhmed** [comment]
@m-toman - you mentioned clipping issues and that's something I'm facing as well. Were you able to track what causes clipping to occur?

**Comment by m-toman** [comment]
@ZohaibAhmed I was able to fix most issues by not using the noam learning rate scheduler but set it to fixed and manually lower it when the loss starts to act funny.
I also found that the simple model in https://github.com/h-meru/Tacotron-WaveRNN behaves much more benign and trains nicer (on 10-bit quantization with "bits", "mulaw" also seems to act up) than the alternative WaveRNN model by fatchord.

**Comment by ZohaibAhmed** [comment]
@m-toman Thanks. I see that in your referred repo, for Tacotron they use Narrow Exponential Decay, and for WaveRNN he sets the learning rate to a fixed number. Which learning rate were you referring to when you say "manually lowering it".

```
#################################################################
        # Narrow Exponential Decay:

        # Phase 1: lr = 1e-3
        # We only start learning rate decay after 50k steps

        # Phase 2: lr in ]1e-5, 1e-3[
        # decay reach minimal value at step 310k

        # Phase 3: lr = 1e-5
        # clip by minimal learning rate value (step > 310k)
        #################################################################
```

**Comment by m-toman** [comment]
WaveRNN - atm I'm starting out with 1e-4 and once the loss starts to act funny, stop it and divide the LR by 10.
Currently training with MoL and there it worked well until 400k steps of batch size 128 until I had to lower it. 
Perhaps "reduce on plateau" or similar would also be a good idea.
